### PR TITLE
Decorate A Class for School library ruby

### PR DESCRIPTION
**Pull request by ismayito**
### Added decorators on the school library
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
#### Summary
______________________________________________________________________________________________________
In this ,I added the following:
- Decorated a class for the school library
- Added the TrimmerDecorecotor to return a name string of 10 characters
- Added a decorator to Capitalize a name string.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,5 +1,45 @@
...
+
+# Decorator class that returns a name string if it's length is not greater than 10 characters
+# otherwise returns the trimmed name
+class TrimmerDecorator < Decorator
+  def correct_name
+    trim_word = @nameable.correct_name.length > 10 ? @nameable.correct_name[0, 10] : @nameable.correct_name
+    puts trim_word
+    trim_word
+  end
+end
```
Comment (DammyShittu) [inline person.rb]: - [ ] Kindly move these classes to their own files. The `person.rb` file should only contain the Person class

File: person.rb [hunk 2]
```diff
@@ -1,5 +1,45 @@
...
+    @nameable.correct_name.capitalize
+  end
+end
+
+# Decorator class that returns a name string if it's length is not greater than 10 characters
+# otherwise returns the trimmed name
+class TrimmerDecorator < Decorator
+  def correct_name
+    trim_word = @nameable.correct_name.length > 10 ? @nameable.correct_name[0, 10] : @nameable.correct_name
+    puts trim_word
```
Comment (DammyShittu) [inline person.rb]: - [ ] Kindly remove this `puts` statement. We should only return the trimmed word.

File: person.rb [hunk 3]
```diff
@@ -1,5 +1,45 @@
...
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+# Decorator class for returning a capitalized name sting
+class CapitalizeDecorator < Decorator
+  def correct_name
+    puts @nameable.correct_name.capitalize
```
Comment (DammyShittu) [inline person.rb]: - [ ] Kindly remove this `puts` statement

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @ismayito,

## Highlights
✔️ Class inheritance feature

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Olamarx — APPROVED** [review]
Hi @ismayito, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Changes fixed ✔️
- Linters passed ✔️
- Meaningful commit messages ✔️

## Optional suggestions
[OPTIONAL] I will encourage you to set up your GETTING STARTED section on your readme file properly so that visitors can understand well.
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**